truss-build/src/truss/build/commands.py

```diff
@@ -0,0 +1,93 @@
+from abc import abstractmethod
+import shlex
+
+
+# TODO(bola): Support secrets
+class Command:
+    @abstractmethod
+    def serialize(self):
+        pass
+
+
+class FromCommand(Command):
+    def __init__(self, image, tag=None, AS=None):
+        self.image = image
+        self.tag = tag
+        self._as = AS
+
+    def serialize(self):
+        ret = f"FROM {self.image}"
+        if self.tag is not None:
+            ret += f":{self.tag}"
+        if self._as is not None:
+            ret += f" AS {self._as}"
+        return ret
```

Comment:
Nit: I know that in this context runtime is not really an issue, but I got used to try to *always* use efficient implementations - unless it's significantly more complex and less readable.

So here you could collect the conditional command in `parts: list[str]` and and the end `" ".join(parts)`.